### PR TITLE
Task structure

### PR DESCRIPTION
**Pull request by fynnfeldpausch**
* updated the task structure
* implemented a more modular design
* improved UI/UX

### CODE REVIEW

File: src/main/webapp/app/tasks/task.service.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if (environment.mockBackend) {
+      return of(this.readTasks());
+    }
+    return of([]); // TODO: Implement http call
+  }
+
+  /**
+   * Adds a new task to the list of tasks.
+   *
+   * @param task the task's name
```
Comment (IamBatman222) [inline src/main/webapp/app/tasks/task.service.ts]: param should be "name"

File: src/main/webapp/app/app.component.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import {Component, OnInit} from '@angular/core';
```
Comment (IamBatman222) [inline src/main/webapp/app/app.component.ts]: You forgot to adapt the unit test

**Comment by fynnfeldpausch** [comment]
I quite like the overflow, to be honest :-D

**Comment by IamBatman222** [comment]
> I quite like the overflow, to be honest :-D

Yes, this one is just a matter of taste. I can also live with the overlapping =)

File: src/main/webapp/app/app.component.ts (hunk 2 above)
Comment (fynnfeldpausch) [reply src/main/webapp/app/app.component.ts]: Added all unit tests except for the task service. We should wait until the HTTP parts are implemented.

**Pull request merged**